src/stash-list/bookmark.vue

```diff
@@ -60,6 +61,11 @@ export default defineComponent({
             return Array.from(related);
         },
 
+        discarded(): boolean {
+            return this.related_tabs.length > 0 && 
+                this.related_tabs.every(t => t.discarded);
```

Comment:
Good point. There was no use of `related_tabs` that didn't want a tab with the same window, so I just added a filter there.

I added the check for hidden tabs.

There's further opportunity for optimization with these changes that aggregate stats about tabs but since I'm trying to keep these pulls independent my options are a bit limited.  I suppose I could add a new pull that does that background stuff and then implement some of these PRs on top of that, but then they couldn't really be accepted independently.

Reply:
Yeah that makes sense.

We can always come back and optimize later, too. :)